### DISCUSSION TITLE
Copilot not working with Python

### DISCUSSION THREAD

**Discussion by Aleksey-Loskutov** [community/community#32587] [Copilot Conversations] [answered]
Hello. After subscribing to GitHub Copilot Monthly plan and paying $10, Copilot doesn't work with Python (in '.py' files). Namely, it does not offer any code options. At the same time, it works well with JavaScript (in '.js' files). Is this a bug? Or should I specify something in the settings? Thanks for any help!

**Comment by davecheney** [comment]
![Screen Shot 2022-09-11 at 13 55 58](https://user-images.githubusercontent.com/7171/189511727-d7ea4fa6-8efb-4568-a2fd-e1b38395583d.png)
What does your Copilot icon look like? If it looks like this, you've disabled support for `.py` files in the past. Click the icon to reactivate it.

Alternatively look in the Copilot extension `settings.json` for this fragment
```
    "github.copilot.enable": {
        "*": true,
        "yaml": true,
        "plaintext": true,
        "markdown": true,
        "go": false
    },
```
Adjust as appropriate.

**Comment by davecheney** [comment]
@Aleksey-Loskutov also, one of your editors is using the expired 1.40 version of the Copilot extension. Please upgrade.

**Comment by Aleksey-Loskutov** [comment]
Thanks for your prompt reply, dear Dave. Copilot is active for `.py ` files. I restored `v1.44.6735` for my Copilot.
Than I checked the Copilot settings on my computer. Here they are (attached, from "C:\Users\User\.vscode\extensions\github.copilot-1.44.6735\package.json" in my Windows ).
But... (!) So... 
I just checked, Copilot is now working fine for `.py` files. This is a little strange, because when I started solving my problem, I had version `v1.44.6735` of Copilot. Has anything been done on your part?
In any case, I am very grateful to you for your help. Thank you very much again!
[package.txt](https://github.com/community/community/files/9542304/package.txt)

**Reply by davecheney** [reply]
Thanks. That’s not the settings file. Go to the extensions panel, choose copilot on the lhs, choose setting on the right then click edit settings.json, you should see something similar to what I posted above.

**Comment by Aleksey-Loskutov** [comment, marked as answer]
Thanks, yes, I found that file too.
I think this discussion can be closed. Can I write my gratitude for your help somewhere on this page? (I'm just a newcomer here :) )
